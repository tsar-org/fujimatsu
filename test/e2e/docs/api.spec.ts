import { createExecutionContext } from 'cloudflare:test';
import worker from '@/../src';
import { IncomingRequest } from 'test/helpers/incomingRequest';
import { mockEnv } from 'test/helpers/mockEnv';
import { describe, expect, it } from 'vitest';

describe('test GET /docs/api', () => {
  it('snapshot test', async () => {
    // arrange
    const request = new IncomingRequest('http://localhost:8787/docs/api');
    const env = mockEnv();
    const ctx = createExecutionContext();

    // act
    const response = await worker.fetch(request, env, ctx);

    // assert
    expect(await response.status).toBe(200);
    expect(await response.text()).toMatchSnapshot();
  });
});
