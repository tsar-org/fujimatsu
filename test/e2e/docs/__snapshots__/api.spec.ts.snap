// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test GET /docs/api > snapshot test 1`] = `
"
      <html lang="en">
        <head>
          <meta charset="utf-8" />
          <meta name="viewport" content="width=device-width, initial-scale=1" />
          <meta name="description" content="SwaggerUI" />
          <title>SwaggerUI</title>
        </head>
        <body>
          
    <div>
      <div id="swagger-ui"></div>
      <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/swagger-ui-dist/swagger-ui.css" />
      <script src="https://cdn.jsdelivr.net/npm/swagger-ui-dist/swagger-ui-bundle.js" crossorigin="anonymous"></script>
      <script>
        window.onload = () => {
          window.ui = SwaggerUIBundle({
            dom_id: '#swagger-ui',url: '/docs/api.json',
          })
        }
      </script>
    </div>
  
        </body>
      </html>
    "
`;
